.github/actions/update-documentation/action.yml

```
@@ -0,0 +1,86 @@
+name: "Deploy Keptn Lifecycle Toolkit on GH cluster"
+description: "Creates a Kind cluster and deploys Keptn Lifecycle Toolkit"
+inputs:
+   version:
+     required: true
+     description: "Version of the Keptn Lifecycle Toolkit Documentation to be deployed"
+   doc-repo:
+     required: true
+     description: "Path to the documentation repository"
+     default: "keptn-sandbox/lifecycle-toolkit-docs"
+   examples-repo:
+     required: true
+     description: "Path to the documentation repository"
+     default: "keptn-sandbox/lifecycle-toolkit-examples"
+   klt-repo:
+     required: true
+     description: "Path to the klt repository"
+     default: "lifecycle-toolkit"
+   update-main:
+     description: "Update the main version of the documentation"
+     required: true
+     default: "false"
+   target-branch:
+     description: "Target branch for the documentation"
+     default: "development"
+     required: true
+   token:
+     description: "Token to access the documentation repository"
+     required: true
+
+
+runs:
+  using: "composite"
+
+  steps:
+  - uses: actions/setup-python@v4
+    with:
+      python-version: '3.10'
+
+  - name: Check out documentation
+    uses: actions/checkout@v3
+    with:
+      repository: ${{ inputs.doc-repo }}
+      path: doc-repo
+      ref: main
+      token: ${{ inputs.token }}
+      fetch-depth: 0
+
+  - name: Check out examples
+    uses: actions/checkout@v3
+    if: ${{ inputs.update-main == 'true' }}
+    with:
+      repository: ${{ inputs.examples-repo }}
+      path: example-repo
+      ref: main
+      token: ${{ inputs.token }}
+      fetch-depth: 0
+
+  - run: pip install -r .github/scripts/doc_update/requirements.txt
+    shell: bash
+
+  - run: python .github/scripts/doc_update/update_docs.py --version ${{ inputs.version }} --klt-docs doc-repo --klt-repo ${{ inputs.klt-repo }} --klt-examples ${{ inputs.examples-repo }} --update-main
+    shell: bash
+    if : ${{ inputs.main-version == 'true' }}
+
+  - run: python .github/scripts/doc_update/update_docs.py --version ${{ inputs.version }} --klt-docs doc-repo --klt-repo ${{ inputs.klt-repo }} --klt-examples ${{ inputs.examples-repo }}
+    shell: bash
+    if: ${{ inputs.main-version != 'true' }}
```

pls add some names to the steps, also i think you can nicely reduce the code here:
```suggestion
  - name: Install dependencies
    run: pip install -r .github/scripts/doc_update/requirements.txt
    shell: bash

  - name: Update documentation
    shell: bash
    run: python .github/scripts/doc_update/update_docs.py --version ${{ inputs.version }} --klt-docs doc-repo --klt-repo ${{ inputs.klt-repo }} --klt-examples ${{ inputs.examples-repo }} --update-main=${{ inputs.main-version }}
```